### PR TITLE
fix check vulnerable nuget packages

### LINKED ISSUE #1596: check vulnerable nuget packages is broken
requires a build first. I will submit a PR

image

https://github.com/getsentry/sentry-dotnet/runs/6173260174?check_suite_focus=true

```
No assets file was found for `/home/runner/work/sentry-dotnet/sentry-dotnet/src/Sentry/Sentry.csproj`.
Please run restore before running this command.

No assets file was found for `/home/runner/work/sentry-dotnet/sentry-dotnet/test/Sentry.Tests/Sentry.Tests.csproj`.
Please run restore before running this command.
```

### PR DESCRIPTION
**Pull request by SimonCropp**
moves vul checking into the build action

fixes #1596 

#skip-changelog

### CODE REVIEW

**Comment by SimonCropp** [comment]
ok i cant get this to work. it always gives

```
Unable to read a package reference from the project `/home/runner/work/sentry-dotnet/sentry-dotnet/src/Sentry/Sentry.csproj`. Please make sure that your project file and project.assets.json file are in sync by running restore.
```
for all non test projects

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -51,6 +51,9 @@ jobs:
               5.0.x
               6.0.x
 
+        - name: Restore
+          run: dotnet dotnet restore Sentry.sln
```
Comment (bruno-garcia) [inline .github/workflows/build.yml:55]: What about?

```suggestion
          run: dotnet restore
```

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -91,3 +95,6 @@ jobs:
             path: |
               ${{ github.workspace }}/src/**/Release/*.nupkg
 
+        - name: List vulnerable packages
```
Comment (bruno-garcia) [inline .github/workflows/build.yml:98]: We've removed building samples and other bits from CI to speed things up. This sounds like a good opportunity to parallelize CI with an extra workflow or job. Is that an option?

(Assuming it adds a minute as you mentioned in a previous chat)

Comment (SimonCropp) [reply .github/workflows/build.yml]: @bruno-garcia @mattjohnsonpint i changed it to be scheduled once a day. thoughts?

File: .github/workflows/vulnerabilities.yml [hunk 3]
```diff
@@ -1,17 +1,25 @@
...
 jobs:
-  check-vulns:
-    runs-on: ubuntu-latest
-
+  List-vulnerable-packages:
+    runs-on: windows-latest
+    env:
+      DOTNET_CLI_TELEMETRY_OPTOUT: 1
+      DOTNET_NOLOGO: 1
+      DOTNET_MULTILEVEL_LOOKUP: 1
```
Comment (mattjohnsonpint) [inline .github/workflows/vulnerabilities.yml:13]: This is the default, according to [this](https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-environment-variables#dotnet_multilevel_lookup).  Though some changes coming for .NET 7.  Is there a reason it needs to be specified here explicitly?

**Comment by bruno-garcia** [comment]
Lgtm. Thanks Simon

**Pull request merged**